src/modules/site-deal/site-deal.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import CustodianConfig from '@ukef/config/custodian.config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { UkefId, UkefSiteId } from '@ukef/helpers';
+
+import { CustodianService } from '../custodian/custodian.service';
+import { CustodianCreateAndProvisionRequest } from '../custodian/dto/custodian-create-and-provision-request.dto';
+import { GraphGetListItemsResponseDto } from '../graph/dto/graph-get-list-item-response.dto';
+import GraphService from '../graph/graph.service';
+import { CreateFacilityFolderRequestItem } from './dto/create-facility-folder-request.dto';
+import { CreateFacilityFolderResponseDto } from './dto/create-facility-folder-response.dto';
+import { SiteDealFolderNotFoundException } from './exception/site-deal-folder-not-found.exception';
+
+type RequiredSharepointConfigKeys = 'tfisFacilityListId' | 'tfisSharepointUrl' | 'scSharepointUrl' | 'scSiteFullUrl' | 'tfisFacilityHiddenListTermStoreId';
+type RequiredCustodianConfigKeys = 'facilityTemplateId' | 'facilityTypeGuid';
+@Injectable()
+export class SiteDealService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly sharepointConfig: Pick<ConfigType<typeof SharepointConfig>, RequiredSharepointConfigKeys>,
+    @Inject(CustodianConfig.KEY)
+    private readonly custodianConfig: Pick<ConfigType<typeof CustodianConfig>, RequiredCustodianConfigKeys>,
+    private readonly graphService: GraphService,
+    private readonly custodianService: CustodianService,
+  ) {}
+
+  async createFacilityFolder(
+    siteId: UkefSiteId,
+    dealId: UkefId,
+    createFacilityFolderRequestItem: CreateFacilityFolderRequestItem,
+  ): Promise<CreateFacilityFolderResponseDto> {
+    const { facilityIdentifier, buyerName } = createFacilityFolderRequestItem;
+
+    const parentFolderName = this.getParentFolderName(buyerName, dealId);
+
+    const facilityFolderName = this.getFacilityFolderName(facilityIdentifier);
+    const parentFolderId = await this.getParentFolderId(siteId, parentFolderName);
+    const termGuid = await this.getTermGuid(facilityIdentifier);
+    const termTitle = facilityIdentifier;
+
+    // TODO apim-139: It's worth noting here that we don't use destinationMarket or riskMarket at all
```

Comment:
Highlighting this todo

Reply:
and for this one, let's removed the unused request fields and document on the ticket and confluence page that we have removed them